src/test/integration/stories/deploy_not_existing_img_test.js

```diff
@@ -0,0 +1,206 @@
+  let logsPage;
+
+  let appName = 'test';
+  let containerImage = 'test';
+  let timeout = 10000;  // 10s
+
+  /**
+   * Waits for element to be present on the page. It doesn't have to be displayed yet.
+   * Additional logic can be given in 'isPresentConditionFn' to f.e. refresh page periodically
+   * until given element appears on the page.
+   * @param {!Element} elm
+   * @param {?function(boolean): boolean} isPresentConditionFn
+   */
+  function waitUntilPresent(elm, isPresentConditionFn) {
+    browser.driver.wait(() => {
+      if (!isPresentConditionFn) {
+        return elm.isPresent();
+      }
+
+      return elm.isPresent().then(isPresentConditionFn);
+    }, timeout);
+  }
+
+  /**
+   * Logic is the same as for waitUntilPresent function with the exception that it
+   * waits for the element to be actually displayed on the page.
+   * @param {!Element} elm
+   * @param {?function(boolean): boolean} isPresentConditionFn
+   */
+  function waitUntilDisplayed(elm, isPresentConditionFn) {
+    browser.driver.wait(() => {
+      if (!isPresentConditionFn) {
+        return elm.isDisplayed();
+      }
+
+      return elm.isDisplayed().then(isPresentConditionFn);
+    }, timeout);
+  }
+
+  beforeAll(() => {
+    // For empty cluster this should actually redirect to zerostate page
+    browser.get('#/replicationcontrollers');
+
+    zeroStatePage = new ZeroStatePageObject();
+    deployPage = new DeployPageObject();
+    replicationControllersPage = new ReplicationControllersPageObject();
+    deleteDialog = new DeleteReplicationControllerDialogObject();
+    replicationControllerDetailPage = new ReplicationControllerDetailPageObject();
+    logsPage = new LogsPageObject();
+  });
+
+  it('should go to deploy page', () => {
+    // when
+    zeroStatePage.deployButton.click();
+    browser.driver.sleep(1000);
+
+    // then
+    expect(browser.getCurrentUrl()).toContain('deploy');
+  });
+
+  it('should deploy app and go to replication controllers list page', () => {
+    // given
+    deployPage.appNameField.sendKeys(appName);
+    deployPage.containerImageField.sendKeys(containerImage);
+
+    // when
+    deployPage.deployButton.click();
+    browser.driver.sleep(1000);
+
+    // then
+    expect(browser.getCurrentUrl()).toContain('replicationcontrollers');
+  });
+
+  it('should wait for card to be in error state', () => {
+    // when
+    waitUntilPresent(replicationControllersPage.cardErrorIcon, (result) => {
+      if (result) {
+        return true;
+      }
+
+      browser.driver.navigate().refresh();
+      browser.driver.sleep(1000);
+      return false;
+    });
+
+    // then
+    expect(replicationControllersPage.cardErrorIcon.isDisplayed()).toBeTruthy();
+    replicationControllersPage.cardErrors.then((errors) => { expect(errors.length).not.toBe(0); });
+  });
+
+  it('should go to details page', () => {
+    // when
+    replicationControllersPage.cardDetailsPageLink.click();
+    browser.driver.sleep(3000);
```

Comment:
Is there a better way to wait for page to fully load before going further with the test? 

waitForAngular doesn't really work.